scenarios/kubernetes_build.py

```diff
@@ -32,32 +32,41 @@ def check(*cmd):
 
 
 def main(args):
-    """Build kubernetes."""
-    env = {}
+    """Build and push kubernetes.
+
+    This is a python port of the kubernetes/hack/jenkins/build.sh script.
+    """
+    env = {
+        # Skip gcloud update checking; do we still need this?
+        'CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK': 'true',
+        # Don't run any unit/integration tests when building
+        'KUBE_RELEASE_RUN_TESTS': 'n',
+    }
+    push_build_args = ['--nomock', '--verbose', '--ci']
     if args.suffix:
-        env['KUBE_GCS_RELEASE_SUFFIX'] = args.suffix
-    env['KUBE_FASTBUILD'] = 'true' if args.fast else 'false'
+        push_build_args.append('--gcs-suffix=%s' % args.suffix)
     if args.federation:
+        # TODO: do we need to set these?
         env['PROJECT'] = args.federation
-        env['FEDERATION'] = 'true'
         env['FEDERATION_PUSH_REPO_BASE'] = 'gcr.io/%s' % args.federation
+        push_build_args.append('--federation')
     if args.release:
-        env['KUBE_GCS_RELEASE_BUCKET'] = args.release
-        env['KUBE_GCS_RELEASE_BUCKET_MIRROR'] = args.release
+        push_build_args.append('--bucket=%s' % args.release)
+    if args.registry:
+        push_build_args.append('--docker-registry=%s' % args.registry)
 
     for key, value in env.items():
         os.environ[key] = value
-
-    check(args.script)
-
+    check('make', 'clean')
```

Comment:
added a check to make sure cwd is `kubernetes`. The `push-build.sh` call will fail if `../release` doesn't exist, so I'm not sure there's much value in an explicit check.